dali/operators/reader/loader/video_loader.cc

```diff
@@ -561,6 +563,24 @@ void VideoLoader::read_file() {
 
       file.last_frame_ = frame;
       key = (pkt->flags & AV_PKT_FLAG_KEY) != 0;
+
+      if (key) {
+        last_key_frame = frame;
+      }
+
+      // if decoding hasn't produced any frames after providing startup_frame_treshold_ frames,
+      // or we are at next key frame
+      if (((key && frame != last_key_frame) || frame > last_key_frame + startup_frame_treshold_)
```

Comment:
`key && frame != last_key_frame`
Is always false.
For `key = false` it's just false, for `key = true` we just set `last_key_frame = frame` so `frame != last_key_frame` is false.

Reply:
Done